### PR TITLE
PUPIL-587 Feat/oak pupil journey layout

### PR DESCRIPTION
**Pull request by benprotheroe**
# How to review this PR

_Leave this text block for the reviewer_

- Check [component hierarchy](https://miro.com/app/board/uXjVNnKBgyk=/?share_link_id=59445593794) is followed correctly
- Check the design [Heuristics](https://components.thenational.academy/?path=/docs/docs-howtodesigncomponents--docs#heuristics-for-component-design) have been followed
- Check [naming conventions](https://components.thenational.academy/?path=/docs/docs-namingconventions--docs) have been applied
- Check for these gotchyas:
  - Missing exports for Oak components
  - Accidental export of Internal components
  - Snapshots of unexpected components have been modified
  - Circular dependencies
  - Code duplication (via not using base components)
  - Non-functional storybook for this or related components
  - Sensitve files changed eg. atoms, or style tokens
  - Relative imports
  - Default exports

# Add your PR description below

PupilJourneyLayout componets

use case: tiers (programme factors), unit, lesson listing

The different sections are passed in as children and props

## Link to the design doc

https://www.figma.com/file/zqpeecgzMGKRli2gUdJOLp/Pupil-Browse?type=design&node-id=4261%3A38021&mode=dev

## A link to the component in the deployment preview

[component link](https://deploy-preview-145--lively-meringue-8ebd43.netlify.app/?path=/docs/components-organisms-pupil-oakpupiljourneylayout--docs)

## Testing instructions

You can view the component using the above link in storybook

test for phases, pages

### CODE REVIEW

**Comment by messmike** [comment]
@benprotheroe looking good, just need to update the height for tablet and mobile breakpoints for the backlink container if possible, e.g:
Screenshot 2024-04-12 at 12 02 21 pm
Screenshot 2024-04-12 at 12 01 55 pm

**Comment by RBrownsett** [comment]
Nice @benprotheroe . Just a couple of things from me:

- The button that takes users back to the previous page seems to be hardcoded with 'View all units' for every section - will this be updated to reflect the correct text (e.g. tier-listing section would display 'Select subject' and unit-listing would display 'Select tier'
image

- Do we need to add any unit or snapshot tests? Noticed we had some in the previous PR so wondered if this was standard practice :-)

**Review by messmike — APPROVED** [review]

File: src/components/organisms/pupil/OakPupilJourneyLayout/OakPupilJourneyLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        $maxWidth={["100%", "all-spacing-22"]}
+        $minWidth={["100%", "all-spacing-22", "all-spacing-23"]}
+        $pt={["inner-padding-none", "inner-padding-m"]}
+        $mb={["space-between-l", "space-between-l", "space-between-xl"]}
+      >
+        <OakFlex $flexDirection={"column"} $gap={["space-between-m2"]}>
+          {titleSlot}
+          <OakHandDrawnHR hrColor={"white"} $height={"all-spacing-1"} />
+        </OakFlex>
+        {children}
```
Comment (kimon-satan) [inline src/components/organisms/pupil/OakPupilJourneyLayout/OakPupilJourneyLayout.tsx:93]: In relation to the problem we discussed earlier about margin: If children was in the above flex box all would be resolved.

**Review by kimon-satan — CHANGES_REQUESTED** [review]

File: src/components/organisms/pupil/OakPupilJourneyLayout/OakPupilJourneyLayout.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+export type OakPupilJourneyLayoutProps = {
+  sectionName: PupilJourneySectionName;
+  topNavSlot: ReactNode;
+  titleSlot: ReactNode;
+  phase: Phase;
+  children: ReactNode;
+};
+
+/**
+ * `OakBox` does not support space-between tokens on `padding` only `margin`, so we need to
```
[the remark this answers is not in the document]
Comment (benprotheroe) [reply src/components/organisms/pupil/OakPupilJourneyLayout/OakPupilJourneyLayout.tsx:24]: removed

**Review by kimon-satan — APPROVED** [review]

**Pull request merged**